tools/awslint/lib/linter.ts

```diff
@@ -126,11 +125,19 @@ export class Evaluation<T> {
     const include = this.shouldEvaluate(this.curr.code, scope);
     const message = util.format(this.curr.message, extra || '');
 
+    // Don't add a "Success" diagnostic. It will break if we run a compound
+    // linter rule which consists of 3 checks with the same scope (such
+    // as for example `assertSignature()`). If the first check fails, we would
+    // add a "Success" diagnostic and all other diagnostics would be skipped because
+    // of the deduplication check above. Changing the scope makes it worse, since
+    // the scope is also the ignore pattern and they're all conceptually the same rule.
+    //
+    // Simplest solution is to not record successes -- why do we even need them?
+    if (include && condition) { return condition; }
```

Comment:
It was mostly for verbose logging, maybe we can emit those here in case we are in verbose mode?